apis/v1beta1/collector_webhook.go

```diff
@@ -20,17 +20,51 @@ import (
 	"strings"
 
 	"github.com/go-logr/logr"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/validation"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
 	"github.com/open-telemetry/opentelemetry-operator/internal/config"
+	ta "github.com/open-telemetry/opentelemetry-operator/internal/manifests/targetallocator/adapters"
```

Comment:
this is unfortunate – I would love it if there were a way out of this conundrum... I don't think that needs to be in this PR but is related to the work I'm doing in #2603. Could you open a ticket and write a TODO here linking it for us to prevent pulling from the manifests package

Reply:
Do you mean moving the methods from [internal/manifests/collector/adapters/config_to_ports.go](https://github.com/open-telemetry/opentelemetry-operator/pull/2861/files#diff-39548fb5f0030535966b3fb437c08814c648216539e297c034f3da6ac5b43a9b) to the API package?

Reply:
yeah... something like that... I think with the new capabilities from the v1beta1 change we really may want to rethink how we can handle all of these adapters.